File: machsuite-posit-rewrite/machsuite-spmv-crs/parse.py
import json

# Rather ugly parsing code

var_names = ["val", "cols", "row_delimiters", "vec", "out"]
in_data = {}
out_data = {}

with open("input.data", "r+") as f:
    idx = 0
    arr = []
    for line_raw in f:
        line = line_raw.strip()
        if line != "%%":
            # print(line)
            arr.append(float(line))
        else:
            if idx != 0:
                in_data[var_names[idx-1]] = {
                "data": arr,
                "format": {
                    "numeric_type": "bitnum",
                    "is_signed": False,
                    "width": 32
                    }
                }
                arr = []
            idx += 1
    in_data["vec"] = {
                "data": arr,
                "format": {
                    "numeric_type": "bitnum",
                    "is_signed": False,
                    "width": 32
                    }
                }


out_data_val = {}
with open("check.data", "r+") as f:    
    idx = 0
    arr = []
    for line_raw in f:
        line = line_raw.strip()
        if line != "%%":
            arr.append(float(line))

    out_data_val = {
                "data": arr,
                "format": {
                    "numeric_type": "bitnum",
                    "is_signed": False,
                    "width": 32
                    }
                }
out_data = in_data.copy()
out_data["out"] = out_data_val
in_data["out"] = out_data_val


############ Write float test data

with open("spmv.in.double.json", "w+") as f:
    f.write(json.dumps(in_data))

with open("spmv.expect.double.json", "w+") as f:
    f.write(json.dumps(out_data))

File: machsuite-posit-rewrite/machsuite-spmv-crs/parse.js
const { transform_machsuite_data_to_json } = require("../../utilities/js/machsuite-data");

const vals = [
    { field: "val", width: 32, dim: [1666], input: true, output: false },
    { field: "cols", width: 32, dim: [1666], input: true, output: false },
    { field: "row_delimiters", width: 32, dim: [495], input: true, output: false },
    { field: "vec", width: 32, dim: [494], input: true, output: false },
    { field: "out", width: 32, dim: [394], input: false, output: true }
];

transform_machsuite_data_to_json("spmv", vals);
